File: app/src/main/java/br/com/flasco/leanworks/data/datasource/remote/checkin/CheckInRemoteDataSource.java
package br.com.flasco.leanworks.data.datasource.remote.checkin;

import javax.inject.Inject;

import br.com.flasco.leanworks.domain.model.CheckIn;
import retrofit2.Call;

public class CheckInRemoteDataSource {
    private final CheckInService checkInService;

    @Inject
    public CheckInRemoteDataSource(CheckInService checkInService) {
        this.checkInService = checkInService;
    }

    public Call<CheckInResponse> saveCheckIn(CheckIn checkIn){
        return checkInService.saveCheckIn(checkIn);
    }
}

File: app/src/main/java/br/com/diogomoreno/events/data/datasource/remote/checkin/CheckInRemoteDataSource.kt
package br.com.diogomoreno.events.data.datasource.remote.checkin

import br.com.diogomoreno.events.domain.model.CheckIn
import retrofit2.Call
import javax.inject.Inject

class CheckInRemoteDataSource @Inject constructor(private val checkInService: CheckInService) {
    fun saveCheckIn(checkIn: CheckIn?): Call<CheckInResponse?>? {
        return checkInService.saveCheckIn(checkIn)
    }

}